src/v/kafka/server/handlers/list_partition_reassignments.cc

```diff
@@ -0,0 +1,173 @@
+
+// Copyright 2020 Redpanda Data, Inc.
+//
+// Use of this software is governed by the Business Source License
+// included in the file licenses/BSL.md
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0
+
+#include "kafka/server/handlers/list_partition_reassignments.h"
+
+#include "cluster/metadata_cache.h"
+#include "cluster/shard_table.h"
+#include "cluster/topics_frontend.h"
+#include "config/node_config.h"
+#include "kafka/protocol/errors.h"
+#include "kafka/protocol/schemata/list_partition_reassignments_request.h"
+#include "kafka/protocol/schemata/list_partition_reassignments_response.h"
+#include "kafka/server/errors.h"
+#include "kafka/server/fwd.h"
+#include "model/namespace.h"
+
+#include <algorithm>
+#include <vector>
+
+namespace kafka {
+
+namespace {
+bool in_broker_list(
+  const model::broker_shard& bshard,
+  const std::vector<model::broker_shard>& list) {
+    return std::find(list.begin(), list.end(), bshard) != list.end();
+}
+} // namespace
+
+// Returns the partition reassignments for a given partition of a topic
+ongoing_partition_reassignment list_partition_reassignments(
+  cluster::partition_assignment& current_assignment,
+  std::optional<std::vector<model::broker_shard>> previous_replica_set) {
+    ongoing_partition_reassignment reassignments;
+
+    std::for_each(
```

Comment:
After I write some code using `std::for_each` i usually remove it because I realize it looks shorter and easier to understand using instead `for(const auto& element : collection) { ... }` haha, no feedback here just chatter

Reply:
I like that philosophy. Lately I've been using `std::for_each` when the loop body does not exceed 5 lines, then use regular `for` loops for the larger bodies.